### PR TITLE
fix(IEC): move iec resource to services/iec and fix codecheck error

### PR DESCRIPTION
**Pull request by corningli**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
./scripts/codecheck.sh ./huaweicloud/services/iec

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                       17      3651      441        32     3178        429           217.70
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~rvices/iec/resource_huaweicloud_iec_vip.go       313       42         9      262         55            20.99
~source_huaweicloud_iec_network_acl_rule.go       323       38         0      285         54            18.95
~rvices/iec/resource_huaweicloud_iec_eip.go       313       38         3      272         53            19.49
~ces/iec/resource_huaweicloud_iec_server.go       593       61        11      521         50             9.60
~ec/resource_huaweicloud_iec_network_acl.go       222       27         2      193         29            15.03
~rce_huaweicloud_iec_security_group_rule.go       221       26         0      195         29            14.87
~iec/resource_huaweicloud_iec_vpc_subnet.go       256       34         3      219         28            12.79
~resource_huaweicloud_iec_security_group.go       219       26         0      193         28            14.51
~rvices/iec/resource_huaweicloud_iec_vpc.go       147       24         1      122         18            14.75
~/data_source_huaweicloud_iec_bandwidths.go       137       21         0      116         14            12.07
~es/iec/resource_huaweicloud_iec_keypair.go       111       16         0       95         14            14.74
~/iec/data_source_huaweicloud_iec_server.go       192       22         3      167         13             7.78
~s/iec/data_source_huaweicloud_iec_sites.go       163       16         0      147         11             7.48
~/iec/data_source_huaweicloud_iec_images.go       112       12         0      100         10            10.00
~iec/data_source_huaweicloud_iec_flavors.go       126       12         0      114         10             8.77
~es/iec/data_source_huaweicloud_iec_eips.go       141       17         0      124          8             6.45
~iec/data_source_huaweicloud_iec_keypair.go        62        9         0       53          5             9.43
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                    17      3651      441        32     3178        429           217.70
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 98190 bytes, 0.098 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
9 iec updateIecVipAssociate huaweicloud/services/iec/resource_huaweicloud_iec_vip.go:258:1
9 iec dataSourceIECServerRead huaweicloud/services/iec/data_source_huaweicloud_iec_server.go:109:1
8 iec resourceIecVIPCreate huaweicloud/services/iec/resource_huaweicloud_iec_vip.go:78:1
7 iec resourceIecServerCreate huaweicloud/services/iec/resource_huaweicloud_iec_server.go:333:1
7 iec waitForSecurityGroupDelete huaweicloud/services/iec/resource_huaweicloud_iec_security_group.go:190:1
7 iec resourceIecSecurityGroupV1Read huaweicloud/services/iec/resource_huaweicloud_iec_security_group.go:120:1
7 iec resourceNetworkACLRuleCreate huaweicloud/services/iec/resource_huaweicloud_iec_network_acl_rule.go:116:1
7 iec waitForIecEipStatus huaweicloud/services/iec/resource_huaweicloud_iec_eip.go:286:1
7 iec resourceIecEipCreate huaweicloud/services/iec/resource_huaweicloud_iec_eip.go:105:1
7 iec dataSourceIecSitesV1Read huaweicloud/services/iec/data_source_huaweicloud_iec_sites.go:94:1
Average: 3.83

==> Checking for golangci-lint...

==> Checking for Nolint directives...
./huaweicloud/services/iec/resource_huaweicloud_iec_vpc.go:138: // lintignore:R018

==> Checking for TF features in iec...
  -> the resource in resource_huaweicloud_iec_network_acl_rule.go should can be imported


==> Checking for misspell in iec...

==> Checking for code complexity in ./huaweicloud/services/acceptance/iec...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                       17      1851      239         0     1612        149           135.71
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~uaweicloud_iec_security_group_rule_test.go       156       21         0      135         25            18.52
~rce_huaweicloud_iec_security_group_test.go       104       16         0       88         16            18.18
~e/iec/resource_huaweicloud_iec_vip_test.go       177       23         0      154         16            10.39
~esource_huaweicloud_iec_vpc_subnet_test.go       147       20         0      127         16            12.60
~e/iec/resource_huaweicloud_iec_vpc_test.go       182       25         0      157         16            10.19
~e/iec/resource_huaweicloud_iec_eip_test.go       109       18         0       91         16            17.58
~ec/resource_huaweicloud_iec_server_test.go       168       15         0      153         16            10.46
~source_huaweicloud_iec_network_acl_test.go       169       20         0      149          6             4.03
~e_huaweicloud_iec_network_acl_rule_test.go       109       14         0       95          5             5.26
~data_source_huaweicloud_iec_images_test.go        53        8         0       45          3             6.67
~c/data_source_huaweicloud_iec_eips_test.go        88       10         0       78          3             3.85
~/data_source_huaweicloud_iec_sites_test.go        54        8         0       46          3             6.52
~ata_source_huaweicloud_iec_flavors_test.go        81       11         0       70          3             4.29
~_source_huaweicloud_iec_bandwidths_test.go        88       10         0       78          3             3.85
~c/resource_huaweicloud_iec_keypair_test.go        69        9         0       60          2             3.33
~ata_source_huaweicloud_iec_keypair_test.go        42        5         0       37          0             0.00
~data_source_huaweicloud_iec_server_test.go        55        6         0       49          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                    17      1851      239         0     1612        149           135.71
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 53833 bytes, 0.054 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
6 iec testAccCheckIecVpcV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vpc_test.go:114:1
6 iec testAccCheckIecVpcSubnetV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vpc_subnet_test.go:77:1
6 iec testAccCheckIecVipExists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vip_test.go:95:1
6 iec testAccCheckIecServerExists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go:56:1
6 iec testAccCheckIecSecurityGroupV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_security_group_test.go:47:1
Average: 2.18

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/iec...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/iec...

==> Cleanup patch...

Check Completed!
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECServerDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec -v -run TestAccIECServerDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECServerDataSource_basic
=== PAUSE TestAccIECServerDataSource_basic
=== CONT  TestAccIECServerDataSource_basic
--- PASS: TestAccIECServerDataSource_basic (264.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       264.322s

make testacc TEST='./huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECSitesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec -v -run TestAccIECSitesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECSitesDataSource_basic
=== PAUSE TestAccIECSitesDataSource_basic
=== CONT  TestAccIECSitesDataSource_basic
--- PASS: TestAccIECSitesDataSource_basic (14.35s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       14.388s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_sites_test.go [hunk 1]
```diff
@@ -42,8 +45,10 @@ func testAccCheckIECSitesDataSourceID(n string) resource.TestCheckFunc {
 	}
 }
 
-var testAccIECSitesConfig string = `
+func testAccIECSitesConfig_basic() string {
+	return `
 data "huaweicloud_iec_sites" "sites_test" {
-  area = "east"
+	area = "east"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_sites_test.go]: set two spaces at the front, not a tab

File: huaweicloud/services/iec/data_source_huaweicloud_iec_server.go [hunk 2]
```diff
@@ -117,58 +119,59 @@ func dataSourceIECServerRead(d *schema.ResourceData, meta interface{}) error {
...
+	allNics, eip := expandIecServerNics(&server)
+	// set volume fields
+	allVolumes, sysDiskID := expandIecServerVolumeAttached(iecClient, &server)
+
+	mErr := multierror.Append(nil,
+		d.Set("name", server.Name),
+		d.Set("status", server.Status),
+		d.Set("flavor_id", server.Flavor.ID),
+		d.Set("flavor_name", server.Flavor.Name),
+		d.Set("image_name", server.Metadata.ImageName),
```
Comment (houpeng80) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_server.go:153]: image_id is missing

Comment (corningli) [reply huaweicloud/services/iec/data_source_huaweicloud_iec_server.go:153]: done

**Pull request merged**